text/0517-io-os-reform.md

```diff
@@ -498,7 +500,65 @@ throughout IO, we can go on to explore the modules in detail.
 ### `std::process`
 [std::process]: #stdprocess
 
-> To be added in a follow-up PR.
+Currently `std::io::process` is used only for spawning new
+processes. The re-envisioned `std::process` will ultimately support
+inspecting currently-running processes, although this RFC does not
+propose any immediate support for doing so -- it merely future-proofs
+the module.
+
+#### `Command`
+[Command]: #command
+
+The `Command` type is a builder API for processes, and is largely in
+good shape, modulo a few tweaks:
+
+* Replace `ToCCstr` bounds with `IntoOsStrBuf`.
+* Replace `env_set_all` with `env_clear`
+* Rename `cwd` to `current_dir`, take `AsPath`.
+* Rename `spawn` to `run`
+* Move `uid` and `gid` to an extension trait in `os::unix`
+* Make `detached` take a `bool` (rather than always setting the
```

Comment:
This form of `detached` actually has a somewhat different semantic meaning than the detached for threads, which is good to point out! For threads you can simply choose at some point during their lifecycle to detach, but for processes there's 2 decisions to make:

1. When spawning, the detached flag affects how the child process is spawned.
2. When dropping a `Child`, whether or not to `wait()` for the child.

I think that the use case of spawning threads and processes is different enough to keep the APIs different, but we may want to explore various names.